x-pack/plugins/profiling/server/routes/setup.ts

```diff
@@ -44,61 +49,64 @@ export function registerSetupRoute({
     async (context, request, response) => {
       try {
         const esClient = await getClient(context);
-        logger.debug('checking if profiling ES configurations are installed');
         const core = await context.core;
-
-        const steps = getProfilingSetupSteps({
-          client: createProfilingEsClient({
-            esClient,
-            request,
-            useDefaultAuth: true,
-          }),
+        const clientWithDefaultAuth = createProfilingEsClient({
+          esClient,
+          request,
+          useDefaultAuth: true,
+        });
+        const setupOptions = {
+          client: clientWithDefaultAuth,
           logger,
           packagePolicyClient: dependencies.start.fleet.packagePolicyService,
           soClient: core.savedObjects.client,
           spaceId: dependencies.setup.spaces.spacesService.getSpaceId(request),
           isCloudEnabled: dependencies.setup.cloud.isCloudEnabled,
-        });
-
-        const hasDataPromise = hasProfilingData({
-          client: createProfilingEsClient({
-            esClient,
-            request,
-          }),
-        });
+        };
 
-        const stepCompletionResultsPromises = checkSteps({ steps, logger });
+        logger.info('Checking if Elasticsearch and Fleet are setup for Universal Profiling');
 
-        const hasData = await hasDataPromise;
+        const state = createDefaultSetupState();
+        state.cloud.available = dependencies.setup.cloud.isCloudEnabled;
 
-        if (hasData) {
-          return response.ok({
+        if (!state.cloud.available) {
+          const msg = `Elastic Cloud is required to set up Elasticsearch and Fleet for Universal Profiling`;
+          logger.error(msg);
+          return response.custom({
+            statusCode: 500,
             body: {
-              has_data: true,
-              has_setup: true,
-              steps: [],
+              message: msg,
             },
           });
         }
 
-        const stepCompletionResults = await stepCompletionResultsPromises;
+        const verifyFunctions = [
+          hasProfilingData,
+          isApmPackageInstalled,
+          validateApmPolicy,
+          validateCollectorPackagePolicy,
+          validateMaximumBuckets,
+          validateResourceManagement,
+          validateSecurityRole,
+          validateSymbolizerPackagePolicy,
+        ];
+        const partialStates = await Promise.all(
+          verifyFunctions.map(async (fn) => await fn(setupOptions))
```

Comment:
I don't think you need to add this `async/await` in the loop, do you? 

What if you just do like:
```suggestion
          verifyFunctions.map(fn) => fn(setupOptions))
```

Reply:
Good catch. I wish the compiler and linter would highlight unnecessary async/await syntax.